### PR TITLE
Use dynamic huffman tables to reduse the crate size.

### PR DESCRIPTION
**Pull request by RazrFalcon**
This commit makes this crate 4x times smaller without performance changes.

Sizes from the `image-png` crate:

Before:
```
% cargo bloat --release --example pngcheck --filter inflate
 File  .text     Size Name
10.4%  70.7% 506.2KiB [1307 Others]
 4.2%  28.7% 205.3KiB inflate::InflateStream::next_state
 0.0%   0.3%   2.0KiB inflate::DynHuffman16::new
 0.0%   0.1%   1.0KiB inflate::InflateStream::run_len_dist
 0.0%   0.1%     508B inflate::DynHuffman16::read
 0.0%   0.1%     475B inflate::InflateStream::update
 0.0%   0.0%     201B inflate::InflateStream::from_zlib
14.7% 100.0% 715.7KiB .text section size, the file size is 4.8MiB
```

After:
```
% cargo bloat --release --example pngcheck --filter inflate
 File  .text     Size Name
10.7%  91.0% 506.2KiB [1307 Others]
 1.0%   8.2%  45.9KiB inflate::InflateStream::next_state
 0.0%   0.4%   2.0KiB inflate::DynHuffman16::new
 0.0%   0.2%   1.0KiB inflate::InflateStream::run_len_dist
 0.0%   0.1%     508B inflate::DynHuffman16::read
 0.0%   0.1%     475B inflate::InflateStream::update
 0.0%   0.0%     201B inflate::InflateStream::from_zlib
11.8% 100.0% 556.3KiB .text section size, the file size is 4.6MiB
```

Benchmark results using [libflate/flate_bench](https://github.com/sile/libflate/tree/master/flate_bench):
```
Before:
-                 inflate: elapsed=1.903052s, size=288632530
After:
-                 inflate: elapsed=1.869871s, size=288632530
```

### CODE REVIEW

**Comment by RazrFalcon** [comment]
Looks like the build is failed due to `?` operator. Should I use `try!`?

**Comment by eddyb** [comment]
@RazrFalcon Yeah, I think so.

File: src/lib.rs [hunk 1]
```diff
@@ -815,36 +813,34 @@ impl InflateStream {
...
+                                    7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, // 256-271
+                                    7, 7, 7, 7, 7, 7, 7, 7, // 272-279
+                                    8, 8, 8, 8, 8, 8, 8, 8, // 280-287
+                                ]));
+                                let dist = try!(DynHuffman16::new(&[
+                                    5, 5, 5, 5, 5, 5, 5, 5,
+                                    5, 5, 5, 5, 5, 5, 5, 5,
+                                    5, 5, 5, 5, 5, 5, 5, 5,
+                                    5, 5, 5, 5, 5, 5, 5, 5
+                                ]));
```
Comment (eddyb) [inline src/lib.rs]: Actually, you should `.unwrap()`, sorry - these can't ever fail, they're effectively constants.

Comment (RazrFalcon) [reply src/lib.rs]: But why it returns 'Result' then?

Comment (eddyb) [reply src/lib.rs]: I meant these calls can't ever fail because they have constant arguments - for other inputs, those functions may return `Err`.

Comment (RazrFalcon) [reply src/lib.rs]: @eddyb done.

**Pull request merged**